pkg/manager/impl/execution_manager.go

```diff
@@ -234,26 +234,36 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	}
 	taskResourceLimits := runtimeInterfaces.TaskResourceSet{}
 
-	// For resource values, we prefer to use the limits set in the application config over the set resource values.
-	if !configResourceLimits.CPU.IsZero() {
+	// For resource values, we prefer to set unset limits to the request values over the config values.
+	if cpuIndex >= 0 {
+		cpuLimit = resourceRequestEntries[cpuIndex].Value
+		cpu, err := resource.ParseQuantity(cpuLimit)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
+		} else {
+			taskResourceLimits.CPU = cpu
+		}
+	} else if !configResourceLimits.CPU.IsZero() {
 		taskResourceLimits.CPU = configResourceLimits.CPU
 	} else {
-		if cpuIndex >= 0 {
-			cpuLimit = resourceRequestEntries[cpuIndex].Value
-		}
 		cpu, err := resource.ParseQuantity(cpuLimit)
 		if err != nil {
 			logger.Errorf(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
```

Comment:
Wont this value be `cpuLimit := "500m"` and if yes do we need to handle the parsing and do error handling . If its const then we can probably have a right quantity set . 

Also the error message might need to change if thats the case

Reply:
talked with Haytham - since we never expect to use these defaults (represents a configuration error if so) we'll log an error when the task limits end up resolving to unset.